tfx/benchmarks/benchmark_base.py

```diff
@@ -27,14 +28,73 @@
     "Beam runner to use - any runner name accepted by "
     "apache_beam.runners.create_runner")
 
+modes = ["default", "local_scaled_execution", "cloud_dataflow", "flink_on_k8s"]
+beam_pipeline_mode = "local_execution"
+num_workers_str = "1"
+cloud_dataflow_temp_loc = ""
+
+def set_beam_pipeline_mode(mode):
+  assert mode in modes
+  globals()['beam_pipeline_mode'] = mode
+
+def set_num_workers(num_workers):
+  globals()['num_workers_str'] = str(num_workers)
+
+def set_cloud_dataflow_temp_loc(temp_loc):
+  globals()['cloud_dataflow_temp_loc'] = temp_loc
```

Comment:
Could we make these instance variables instead of globals?

You'd also have to make these methods instead of global functions, and possibly initialize the instance variables to `None` in the `__init__`. Alternatively, you could subclass `BenchmarkBase` to a new subclass to preserve the backwards compatibility of `BenchmarkBase`.

Reply:
I think to make them instance variables and methods I'd have to plumb the execution parameters through the tfma_benchmark_base.py, tfma_v2_benchmark_base.py, and tft_benchmark_base.py when they're first created since the beam pipeline is created via BenchmarkBase._create_beam_pipeline(). Let me know if that's the approach you want me to take.

Reply:
`TFMABenchmarkBase`, `TFMAV2BenchmarkBase` and `TFTBenchmarkBase` all subclass from `BenchmarkBase`, so you can make `set_beam_pipeline_mode` a method on `BenchmarkBase` and it would be accessible from those classes.

For example, in `_run_tfma_benchmarks()` you could then call `set_beam_pipeline_mode()` after constructing `TFMABenchmarkBase`. `_create_beam_pipeline()` can then read from this instance variable instead of from the global.